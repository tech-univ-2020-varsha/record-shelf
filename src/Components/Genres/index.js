import React from 'react';
import GenreCard from '../GenreCard';
import * as styles from './index.module.css';

const Genres = () => (
  <div>

    <div className={styles.genreContainer}>
      <div className={styles.genreText}>
        genres
      </div>
      <div className={styles.genres}>
        <div className={styles.card}>
          <GenreCard cardName="rock" />
        </div>
        <div className={styles.card}>
          <GenreCard cardName="pop" />
        </div>
        <div className={styles.card}>
          <GenreCard cardName="bollywood" />
        </div>
        <div className={styles.card}>
          <GenreCard cardName="country" />
        </div>
      </div>
    </div>
  </div>
);


export default Genres;
