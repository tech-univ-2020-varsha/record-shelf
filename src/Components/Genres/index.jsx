import React from 'react';
import propTypes from 'prop-types';
import GenreCard from '../GenreCard';
import * as styles from './index.module.css';

const Genres = (props) => {
  const {
    rockData, popData, bollywoodData, countryData,
  } = props.location.state;
  return (

    <div className={styles.genreContainer}>
      <div className={styles.genreText}>
        genres
      </div>
      <div className={styles.genres}>
        <div className={styles.row}>
          <div className={styles.card}>
            <GenreCard cardName="rock" genreDetails={rockData} />
          </div>
          <div className={styles.card}>
            <GenreCard cardName="pop" genreDetails={popData} />
          </div>
        </div>
        <div className={styles.row}>
          <div className={styles.card}>
            <GenreCard cardName="bollywood" genreDetails={bollywoodData} />
          </div>
          <div className={styles.card}>
            <GenreCard cardName="country" genreDetails={countryData} />
          </div>
        </div>
      </div>
    </div>

  );
};


export default Genres;

Genres.propTypes = {
  rockData: propTypes.string.isRequired,
  popData: propTypes.string.isRequired,
  bollywoodData: propTypes.string.isRequired,
  countryData: propTypes.string.isRequired,

};
