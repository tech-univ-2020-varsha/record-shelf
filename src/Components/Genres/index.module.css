.genres{
    margin: 0;
    height: 100%;
    display: flex;
  flex-wrap: wrap;
    /* align-items: center; */
    justify-content: space-evenly;
}

.card{
    width: 100%;
    background-color: rgb(215,215,215);
    border-radius: 10%;
    padding: 2em;
    margin: 2em;
}

.genreContainer{
    background-color: rgb(245,245,245);
    margin: 0 6em 0 6em ;
    height: 100%;
    width: 100%;
}
.genres>div{
    flex: 0 50%;
}


.genreText{
    color: black;
    font-family:Poppins;
    font-weight: bold;
    font-size: 3em;
}