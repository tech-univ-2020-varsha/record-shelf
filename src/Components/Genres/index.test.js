import { render } from '@testing-library/react';
import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Genres from '.';

describe('the genres', () => {
  it('should render correctly', () => {
    const history = createMemoryHistory();

    const { asFragment } = render(
      <Router history={history}><Genres /></Router>,
    );
    expect(asFragment).toMatchSnapshot();
  });
});
