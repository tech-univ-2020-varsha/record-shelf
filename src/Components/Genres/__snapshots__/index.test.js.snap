// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the genres should render correctly 1`] = `[Function]`;
